lib/ios.js

```diff
@@ -0,0 +1,130 @@
+import {cp, fs, macOsxVersion, ok, nok} from './utils';
+import {DoctorCheck} from './doctor';
+import log from './logger';
+import path from 'path';
+import NodeDetector from './node-detector';
+let checks = [];
+
+// Helpers
+
+async function authorizeIos () {
+  try {
+    var authorizePath = path.resolve(__dirname, "../../bin", "authorize-ios.js");
+    cp.exec("'" + process.execPath + "' '" + authorizePath + "'", { maxBuffer: 524288});
+  } catch (err) {
+    throw new Error('Could not authorize iOS: ' + err);
+  }
+}
+
+// Check for Xcode.
+class XcodeCheck extends DoctorCheck {
+ async diagnose() {
+    let xcodePath;
+    const errMess = 'Xcode is NOT installed!';
+    try {
+      let [stdout] = await cp.exec("xcode-select --print-path", { maxBuffer: 524288});
+      xcodePath = stdout.replace("\n", "");
+    } catch (err) {
+      log.debug(err);
+      return nok(errMess);
+    }
+    return xcodePath && await fs.existsSync(xcodePath) ? ok(`Xcode is installed at {xcodePath}.`) : 
+      nok(errMess);
+  }
+
+  async fix () {
+    await cp.exec("xcode-select --install", { maxBuffer: 524288});
+  }
+}
+checks.push(new XcodeCheck());
+
+// Check for Xcode Command Line Tools.
+class XcodeCmdLineToolsCheck extends DoctorCheck {
+  async diagnose () {
+    const errMess = "Xcode Command Line Tools are NOT installed!";
+    let pkgName = await macOsxVersion() === '10.8' ? "com.apple.pkg.DeveloperToolsCLI" : "com.apple.pkg.CLTools_Executables";
```

Comment:
Yeah I will get rid of stuff/move thing around after I get the first version working.